Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -624,12 +630,45 @@ void CGPathApply(CGPathRef path, void* info, CGPathApplierFunction function) {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 bool CGPathContainsPoint(CGPathRef path, const CGAffineTransform* m, CGPoint point, bool eoFill) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (m) {
+        point = CGPointApplyAffineTransform(point, *m);
+    }
+    // check if the point is outside this box already, if it is, return false
+    CGRect boundingBox = CGPathGetBoundingBox(path);
+
+    if ((boundingBox.origin.x > point.x) || (boundingBox.origin.y > point.y) ||
+        ((boundingBox.origin.x + boundingBox.size.width) < point.x) || ((boundingBox.origin.y + boundingBox.size.height) < point.y)) {
```

Comment:
Updated.

On Jun 28, 2016, at 1:07 PM, Ben Viglietta <notifications@github.com<mailto:notifications@github.com>> wrote:


In Frameworks/CoreGraphics/CGPath.mm<https://github.com/Microsoft/WinObjC/pull/587#discussion_r68832605>:

>   @Notes
>  */
>  bool CGPathContainsPoint(CGPathRef path, const CGAffineTransform* m, CGPoint point, bool eoFill) {
> -    UNIMPLEMENTED();
> -    return StubReturn();
> +    if (m) {
> +        point = CGPointApplyAffineTransform(point, *m);
> +    }
> +    // check if the point is outside this box already, if it is, return false
> +    CGRect boundingBox = CGPathGetBoundingBox(path);
> +
> +    if ((boundingBox.origin.x > point.x) || (boundingBox.origin.y > point.y) ||
> +        ((boundingBox.origin.x + boundingBox.size.width) < point.x) || ((boundingBox.origin.y + boundingBox.size.height) < point.y)) {


This can be simplified somewhat:

if (!CGRectContainsPoint(boundingBox, point))

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Microsoft/WinObjC/pull/587/files/83c35ea7f13358471f0a3f5483315396816db73e#r68832605>, or mute the thread<https://github.com/notifications/unsubscribe/AJC9Ajr9ZBK3SX3K6tel4SLKRPVxNJ4_ks5qQX7-gaJpZM4I_fp1>.